### ISSUE TITLE
make setup.py and admin/sumo-setup.py share code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11521]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1479|
|Type|defect|
|Created|2006-02-12 19:07:39Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1479 1479
type__defect defect
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1139771259000000 1139771259000000
changetime__1147625851000000 1147625851000000
version__ 
owner__ 
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
This is my own stupid fault: I forked setup.py and admin/sumo-setup.py when I
changed the sumo tarball to contain subdirectories that are basically the
released form of each subproject instead of looking just like an SVN checkout.
So Jp's recent changes to setup.py aren't taking effect for newly created sumo
releases. I need to make them share code before I can put out a sumo.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
I just copied the fixes exarkun made to sumo-setup.py for this release. Moving
to bug prio.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

This has been implemented in [branches/release-versioning-1668,](branches/release-versioning-1668,) which is where [#11201](https://github.com/twisted/twisted/issues/11201) is also being worked on. Not quite ready for review, though.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set status to `closed`|
|-|-|

This was resolved along with [#11201](https://github.com/twisted/twisted/issues/11201) in r16792.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
